src/Umbraco.Web/Models/Mapping/SensitivePropertiesResolver.cs

```diff
@@ -0,0 +1,47 @@
+﻿using System.Collections.Generic;
+using System.Linq;
+using AutoMapper;
+using Umbraco.Core.Models;
+using Umbraco.Core.Services;
+using Umbraco.Web.Models.ContentEditing;
+
+namespace Umbraco.Web.Models.Mapping
+{
+    internal class SensitivePropertiesResolver<TSource> : IValueResolver where TSource : IContentBase
+    {
+        public ResolutionResult Resolve(ResolutionResult source)
+        {
+            if (source.Value != null && (source.Value is TSource) == false)
+                throw new AutoMapperMappingException(string.Format("Value supplied is of type {0} but expected {1}.\nChange the value resolver source type, or redirect the source value supplied to the value resolver using FromMember.", new object[]
+                {
+                    source.Value.GetType(),
+                    typeof (TSource)
+                }));
+            return source.New(
+                //perform the mapping with the current umbraco context
+                ResolveCore(source.Context.GetUmbracoContext(), (TSource)source.Value), typeof(IEnumerable<ContentPropertyDisplay>));
+        }
+
+        protected virtual IEnumerable<ContentPropertyDisplay> ResolveCore(UmbracoContext umbracoContext, TSource content)
```

Comment:
This result isn't quite correct for this mapping. The list view doesn't use `ContentPropertyDisplay`, it only uses `ContentPropertyBasic`. This is working currently because `ContentPropertyDisplay` inherits from the basic class, but if we did actually want to re-use this class for content/media list views if they ever supported sensitive data, we'd want to make sure we're dealing with the correct classes.